src/core/json/jsonpath_grammar.y

```diff
@@ -76,21 +80,28 @@ opt_relative_location:
 
 relative_location: DOT relative_path
         | DESCENT { driver->AddSegment(PathSegment{SegmentType::DESCENT}); } relative_path
-        | LBRACKET bracket_index RBRACKET { driver->AddSegment($2); } opt_relative_location
+        | bracket_expr
 
 relative_path: identifier { driver->AddIdentifier($1); } opt_relative_location
         | WILDCARD { driver->AddWildcard(); } opt_relative_location
-
+        | bracket_expr
 
 identifier: UNQ_STR
-         // | single_quoted_string | double_quoted_string
 
-bracket_index: WILDCARD { $$ = PathSegment{SegmentType::INDEX, IndexExpr::All()}; }
+bracket_expr: LBRACKET bracket_key RBRACKET { driver->AddSegment($2); } opt_relative_location
+
+bracket_key: single_quoted_string { $$ = PathSegment(SegmentType::IDENTIFIER, $1); }
```

Comment:
I'm not sure that `bracket_key` is the best name for this nonterminal. Open for suggestions

Reply:
Bracket index was an ok name, both strings and integers can be "index"... Suggestions would be bracket_inner, bracket_core 🤷🏻‍♂️